src/sugar.js

```diff
@@ -0,0 +1,35 @@
+import { Monoid } from 'funcadelic';
+import { parameterized } from '../src/types';
+
+const { keys } = Object;
+
+let ContainsTypes = Monoid.create(class ContainsTypes {
+  empty() { return true; }
+  append(a, b) {
+    // this is necessary because Funcadelic passes different arguments
+    // for arrays than objects. For arrays, value is just the value and 
+    // for Objects it's Monoidic with {value}. 
+    // I think it's a bug in Funcadelic.
+    let value = b && b.value || b;
```

Comment:
As discussed in As discussed in https://github.com/cowboyd/funcadelic.js/issues/18, we should have a typeclass to convert values into lists if needed.

Reply:
done